db/migrate/20240328075919_add_trial_ended_at_to_subscriptions.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class AddTrialEndedAtToSubscriptions < ActiveRecord::Migration[7.0]
+  class Subscription < ApplicationRecord
+    belongs_to :customer
+    belongs_to :plan
+
+    # NOTE: We reimplement the logic from Subscription#initial_started_at  differently to avoid N+1 queries
+    #       eager loading subscription.customer.subscriptions is not enough.
+    def initial_started_at
+      customer.subscriptions.select do |s|
+        s.external_id == external_id && s.started_at.present?
+      end.min_by(&:started_at)&.started_at || subscription_at
+    end
+  end
+
+  class Customer < ApplicationRecord
+    has_many :subscriptions
+  end
+
+  class Plan < ApplicationRecord
+    has_many :subscriptions
+  end
+
+  def up
+    add_column :subscriptions, :trial_ended_at, :datetime
+
+    Subscription
+      .joins(:plan)
+      .where(trial_ended_at: nil)
+      .where.not(plans: { trial_period: nil })
+      .includes(:plan, customer: :subscriptions)
+      .find_each do |subscription|
+        trial_ended_at = subscription.initial_started_at.to_date + subscription.plan.trial_period.days
```

Comment:
In this case, I don't think we need to look at timezones because the migration:
* reads UTC date in the db
* add duration
* store UTC dates in the db
It's independent of "when the migration is run" unlike the `BillingService` or `FreeTrialBillingService`.

Worth case scenario if I overlooked something, this migration is here to populate the database for consistency but we only use this timestamp as a boolean for now. I think if the date is offset by a few hours it's acceptable.

Reply:
The issue I see with is that  `subscription.initial_started_at.to_date` is rounding down the `started_at` to the beginning of the day (in UTC), meaning you are not keeping the time precision. Depending on the customer timezone, it could change the day (+/- 1 day).
I would suggest to get rid of this `.to_date` to keep the UTC timestamp

Reply:
You're right! Excellent catch 🙏
I have no idea why I added a `to_date` here 🤔 

For future reference: the data migration as removed from this PR before merging.